digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/queries.scala

```diff
@@ -283,8 +313,11 @@ object Queries {
 
   def getEnumeratedPropertyValues(assetTypeId: Long): Seq[EnumeratedPropertyValue] = {
     Q.query[Long, EnumeratedPropertyValueRow](enumeratedPropertyValues).apply(assetTypeId).list.groupBy(_.propertyId).map { case (k, v) =>
-      val row = v(0)
-      EnumeratedPropertyValue(row.propertyId, row.propertyPublicId, row.propertyName, row.propertyType, row.required, v.map(r => PropertyValue(r.value.toString, Some(r.displayValue))).toSeq)
+      val row = v.head
+      EnumeratedPropertyValue(row.propertyId, row.propertyPublicId, row.propertyName, row.propertyType, row.required, v.map(r => {
+        val value: AnyVal = if (r.value % 1 == 0) r.value.toInt else r.value
+        PropertyValue(value.toString , Some(r.displayValue))
+      }))
```

Comment:
Why we fist validate if the value is an Int or a Double and then stored like a String?
For me, that validation doesn't make sense since that is the same as:
```Scala
PropertyValue(r.value.toString , Some(r.displayValue))

Reply:
This validation is here to make a difference when representing values like 4.0 to 4 since now we have to treat this parameter as a Double type instead of an Int